lib/main.ml

```diff
@@ -87,8 +87,8 @@ let file =
     ~doc:"Configuration file for Mirage. If not specified, the current directory will be scanned. \
           If one file named $(b,config.ml) is found, that file will be used. If no files \
           or multiple configuration files are found, this will result in an error unless one \
-          is explicitly specified on the command line." [] in
-  Arg.(value & pos 0 (some string) None & doc)
+          is explicitly specified on the command line." ["f"; "file"] in
```

Comment:
Ok, so the situation is as follows:

- We need to compile the `config.ml` (or similar) in order to be able to construct the cmdliner term.
- We can't use `Cmdliner.eval` to extract the config file name because we haven't constructed the full cmdliner term yet.
- Right now I am using `Term.eval_peek_opts` to extract the config file name but that only works with optional parameters.
Indeed, there is no way to know if something is a positional parameter when one does not know what the optional parameters are: i.e., `cmd -X Y` could be parsed as either an optional parameter `X` with value `Y` or a flag `X` with positional parameter `Y`.

I don't see an easy way to work around this.  Any ideas?